### ISSUE TITLE
Problem with overriding a plugin element when an Admin prefix is used

### ISSUE THREAD

**Issue by infoliv** [cakephp/cakephp#16982]
### Description

My App is both using prefix `Admin` and having a plugin `RubriquesManager`.

In the plugin `RubriquesManager` I have 2 elements `articles` for each side (front side and admin side) :

- `/plugins/RubriquesManager/templates/element/articles.php` displays articles for front side

- `/plugins/RubriquesManager/templates/Admin/element/articles.php` displays articles for admin side (with buttons for article editing, deleting, ...)

In order to override in my app the  front side element `articles` of plugin `RubriquesManager` I have created the file :

`/templates/plugin/RubriquesManager/element/articles.php`

The problem is that Cake is using this element both for overriding element `articles` for front side and also for admin side while I didn't create `/templates/plugin/RubriquesManager/Admin/element/articles.php` !...

Is there a way to prevent Cake from trying to override the admin side element  with the front one ?

https://stackoverflow.com/questions/75262721/problem-with-overriding-a-plugin-element-when-an-admin-prefix-is-used-in-cakephp?noredirect=1#comment132808705_75262721

### CakePHP Version

4.4.10

**Comment by markstory** [comment]
I don't think that is possible today. I've personally never used plugins + prefixes + app override templates, and I don't recall any issues about this in the past. The `View::_paths()` method and `View::_getTemplateFileName()` methods are the relevant ones here. Have you taken a look at those methods and what paths they consider in this situation?

**Comment by infoliv** [comment]
As ndm mentionned in his stackoverflow comment, the current order is :
1. app prefix-element,
2. app element,
3. plugin prefix-element,
4. plugin element.

I would expect the order :
1. app prefix-element,
2. plugin prefix-element,
3. app element,
4. plugin element.

**Comment by markstory** [comment]
Thank you for clarifying the current expected order. It is helpful having all of the context for an issue in one place.

My concern with changing the template ordering is that it has a possibility of disrupting existing applications. No longer would app override templates be used for plugin prefix elements. While this change would be opening up new usage, the change in behavior could be interpreted as a bug, as what worked before no longer works. One possibility is to change the order in the upcoming 5.0 release.

**Comment by asgraf** [comment]
I would like to point that there is at least one CakePHP plugin i know of that interferes with those paths and should be checked for compatibility with changed order:
`FriendsOfCake/crud-view`

**Comment by infoliv** [comment]
Hello github bot, could you please remove the stale label ?

**Comment by infoliv** [comment]
Could this stale label be removed please ?

**Comment by infoliv** [comment]
Could this stale label be removed please ?

**Comment by dereuromark** [comment]
Almost 3 years later there is now a possible fix for this :)

**Comment by markstory** [comment]
Closing this as #19034 closed
